### DISCUSSION TITLE
Agenda workshop May 2024

### DISCUSSION THREAD

**Discussion by axel-lauer** [ESMValGroup/Community#162] [Announcements]
Hi all @ESMValGroup/esmvaltool-developmentteam,

Here is the first draft agenda for the next ESMValTool workshop. Comments and feedback very welcome! Please let us know whether you would be fine organizing and chairing a session and/or preparing a contribution for a session. We started suggesting a few "volunteers", please let us know whether you are fine with this or not.

# Agenda ESMValTool Community Workshop 2024 (hybrid meeting)
**(all times are given in CEST)**

### 27 May 2024 - Day 1 (Monday)
9.30 am – 9.45 am 	**Introduction, welcome and workshop agenda** (Birgit)
9.45 am – 10.15 am 	**ESMValTool v2.11.0 release** (Emma, Chris, Ed)
10.15 am – 11.00 am 	**Getting ready for CMIP7** (Romain)
  * On-the-fly CMORizer for ACCESS (Romain)
  * CMIP7 development and submission including training of users and identification of gaps in our evaluation workflow (Romain)
  * Development of a workflow at ACCESS NRI that uses ESMValTool (Romain)

11.00 am – 11.15 am 	**Coffee break**
11.15 am – 12.00 pm 	**Getting ready for CMIP7** (Alistair)
  * Update on the ESMValTool application/activities within the Fresh Eyes community (Evgenia)
  * ICONEval and model benchmarking (Manuel, Axel)
  * General discussion (tbd)

12.00 pm – 01.00 pm 	**Lunch break**
01.00 pm – 01.30 pm 	**Introduction of possible coding sessions, e.g.** (Bouwe)
  * Implementing supporting for different versions of obs datasets (Axel) (see https://github.com/ESMValGroup/ESMValTool/issues/3452)
  * Improved ERA5 support (derived variables, grib) (Manuel, Bouwe)
  * Starting on new configuration file (see https://github.com/ESMValGroup/ESMValCore/issues/2371, https://github.com/ESMValGroup/ESMValCore/issues/2369) (Manuel)
  * Tutorial extension (Ranjini)
  * Website updating (Axel)
  * Support for station data (Birgit)
  * Porting of selected obs datasets from tier3 to tier2 (Axel) (see https://github.com/ESMValGroup/ESMValTool/issues/3331)
  * Cleaning up the repository (e.g. deleting old branches) (tbd)
  * An introduction to using dask with some practice/demo (Bouwe)
  * Recipe Test Workflow - code review and adding new recipes.

01.30 pm – 03.30 pm 	**Coding session** (all)
03.30 pm – 03.45 pm	**Coffee break**
03.45 pm – 04.45 pm	**Coding session** (all)
04.45 pm – 05.00 pm	**Summary and wrap up of first day** (Birgit)

### 28 May 2024 - Day 2 (Tuesday)
9.30 am – 9.40 am 	**Introduction day 2 and workshop agenda** (Birgit)

9.40 am – 10.15 am 	**Ocean diagnostics** (Alistair)
  * Porting ocean diagnostics from the Consortium for Ocean and Sea-Ice Modeling in Australia (COSIMA) to ESMValTool (Romain)
  * Recipe for ENSO evaluation (Romain)
  * Challenges of dealing with ocean unstructured grids and tripolar grid in NEMO. How difficult will it be to work with ocean grids in ESMValTool? (User Engagement Team, tbd)
  * Where does it make sense to use x-array? (Katja, User Engagement Team)

10.15 am – 11.00 am 	**Technical enhancements** (Manuel, Bouwe, V)
  * Status on updating / maintenance of software dependencies including new Python version (V)
  * Reading and processing data from intake catalogs, using the intake-esm library (Bouwe, Katja and Manuel about use case)
  * Re-organizing our configuration files https://github.com/ESMValGroup/ESMValCore/issues/2371, e.g. using different Dask configurations, different DRS for different rootpaths, etc.; new configuration file plan (Bouwe)

11.00 am – 11.15 am 	**Coffee break**
11.15 am – 12.00 am 	**Technical enhancements** (Manuel, Bouwe)
  * Read and processing data in Zarr format; (what) do we need to read model data on HEALPix grids? (often intake and Zarr)? (Manuel, Katja)
  * High-resolution model data: often the weights to interpolate grids can be reused, not only for different variables but even across different experiments, time periods, and simulations of the same model. Could ESMValTool support the reading and use of precalculated weights because their calculation is very time-consuming? (Katja)
  * Presentation on how ECCC operationalizes ESMValTool for extreme event attribution, discussion on recipes generated by Python scripts (Liza)

12.00 pm – 01.00 pm 	**Lunch break**
01.00 pm – 01.10 pm 	**Preparation for coding sessions** (Alistair)
01.10 pm – 03.00 pm 	**Coding session** (all)
03.00 pm – 03.15 pm 	**Coffee break**
03.15 pm – 05.00 pm 	**Coding session** (all)
05.00 pm – 05.15 pm 	**Summary and wrap up of second day** (Birgit)

06.30 pm 	**Workshop dinner (self paid)**

### 29 May 2024 - Day 3 (Wednesday)
9.30 am – 9.40 am 	**Introduction day 3 and workshop agenda** (Birgit)
9.40 am – 10.00 am 	**Discussion: ESMValTool long-term strategy** (Alistair)
10.00 am – 10.30 am 	**Tutorial extension** (Bouwe)
  * Developing teaching material (Felicity)
  * Extending the Python API (tbd)
  * Jupyter notebooks (Ranjini)
  * Configuring Dask for use with ESMValTool (Bouwe)

10.30 am – 11.00 am 	**General discussions** (Axel)
  * Guidelines for proposal contributions (Ranjini)
  * Cleaning up inactive users (Axel) (see https://github.com/ESMValGroup/Community/discussions/151)

11.00 am – 11.15 am 	**Coffee break**
11.15 am – 12.00 pm 	**Observational data** (Bouwe)
  * Observation data management, access to observational data is important to facilitate using ESMValTool: there have been difficulties in keeping observation data updated across platforms (Ranjini) (see https://docs.google.com/document/d/1P3QcODeu82IsakYwBUXbzytk8xdRV2_-PlMRBzmN4w0/edit)
  * Support for station data, status and open issues (Alistair, Birgit)
  * Porting selected obs datasets from tier3 to tier2 (Axel, Bouwe) (see https://github.com/ESMValGroup/ESMValTool/pull/3624)

12.00 pm – 01.00 pm 	**Lunch break**
01.00 pm – 01.10 pm 	**Preparation for coding sessions** (Birgit)
01.10 pm – 03.00 pm 	**Coding session** (all)
03.00 pm – 03.15 pm 	**Coffee break**
03.15 pm – 05.00 pm 	**Coding session** (all)
05.00 pm – 05.15 pm 	**Wrap up and future workshops** (Birgit)

**Comment by schlunma** [comment]
Thanks Axel, looks good to me! I removed most the question marks from my name. Regarding the use of intake: I can introduce a use case (maybe with @katjaweigel), but I am not at all familiar with the technical details.

**Reply by katjaweigel** [reply]
Thanks for the agenda! I got some infos and data to use intake here (outside ESMValTool), I hope I manage to look at it before the workshop.

**Reply by bouweandela** [reply]
I could help out with the technical details

**Comment by egalytska** [comment]
Hi @axel-lauer, not sure how important it is, but  I could provide a brief update on the ESMValTool application/activities within the Fresh Eyes community.

**Reply by axel-lauer** [reply]
Sounds good to me, maybe as a contribution to the session "Getting ready for CMIP7" on Monday morning. Does that sound good?

**Reply by egalytska** [reply]
Sure, sounds great.

**Comment by valeriupredoi** [comment]
@axel-lauer you guys can use me for anything you please :grin: But it's gotta be technical or bier/panzer/rennsport-related, I don't speak science :beer:

**Reply by axel-lauer** [reply]
With pleasure. Is there any technical topic that you would be fine introducing or chairing? Or maybe another technical topic that's not on the list yet?

**Reply by valeriupredoi** [reply]
can't think of anything just now, bud - but if you or others suggest anything I'd be happy to oblige :beers:

**Reply by axel-lauer** [reply]
I volunteered you for "status on updating / maintenance of software dependencies including new Python version" in our Tuesday session "10.15 am – 11.00 am 	**Technical enhancements** (Manuel, Bouwe, V)". I hope that's fine, otherwise please let me know.

**Reply by valeriupredoi** [reply]
more than fine, cheers muchly, Axel :beer:

**Comment by axel-lauer** [comment]
@ESMValGroup/esmvaltool-developmentteam 

If you haven't already done so and you are planning on attending the upcoming ESMValTool workshop (27-29 May 2024) in person or online, please register by Wednesday, **8 May 2024** at the latest. Thank you,

The workshop registration (online or in person) is open until 8 May 2024: https://forms.gle/7sDF5mL7BVP2DqoW6

**Comment by bouweandela** [comment]
@axel-lauer I replaced some `tbd` with my name, but I did not want to take all of them to also give others an opportunity to do this. However, if you cannot find anyone, I would be happy to chair/introduce more topics if needed.

**Comment by malininae** [comment]
@axel-lauer I updated a bit the part on the presentation I want to give. Also, I was hoping that I will have a few PRs to clean up but I don't have to many :flushed:

**Comment by rbeucher** [comment]
Hi Everyone,

I won't be able to join in person unfortunately. However, I plan to attend the morning session on Monday. Being 8 hours ahead of Munich, my participation will depend on how I feel after a full day of work. Nonetheless, I'm eager to discuss our CMIP7 plans.

We have a workshop event in Canberra on Tuesday and Wednesday, which is not related to ESMValTool. I will try to join part of the morning session on Tuesday, but I won't be able to attend the Wednesday morning session.

Are there plans to record the sessions? I would be interested in watching them...

@flicj191 and @headmetal will be attending and representing ACCESS-NRI. Felicity has prepared a short presentation on the Hackathon we held last March in Melbourne. Mike and Felicity have been collaborating with the Australian community and can share our general approach for ESMValTool.

## ESMValCore

I believe ESMValCore already offers most of what researchers need for their analysis. Key areas we plan to contribute to include:

- Efficient loading of 3D and high-resolution datasets, especially for ocean evaluation
Handling of ocean model grids (for NEMO and MOM)
- Improving the Python API for better integration into Jupyter notebooks, enabling interactive development of recipes and diagnostics
- Dask optimization, lazy loading for all preprocessors
- Xarray improvements

ESMValCore is an excellent tool, and I think it would benefit from better documentation and dedicated tutorials on using the Python API. Optimization is crucial as we often work with larger datasets.

## ESMValTool

Our community finds the ESMValTool learning curve quite steep, so any teaching materials that can help with this are welcome. We've started a YouTube series and plan to add more content.

ESMValTool is essentially a collection of diagnostics and recipes developed by the community. Over the past six months, I've found it challenging to identify suitable tools for specific cases.

ACCESS-NRI plans to adapt and optimize a subset of recipes, focusing on ACCESS model evaluation to support our CMIP7 development and submission. The community has also requested simpler recipes to build from.

There are a few issues we should discuss, as they seem burdensome for developers and maintainers. These include dataset deprecation, support for languages like NCL, CMIP3 support, and others. Additionally, the current model makes it difficult to contribute new recipes/diagnostics and could be streamlined. The size of the ESMValTool conda environment, with its many dependencies, also makes it heavy to deploy.

Our team is looking forward to participating in the discussions, and we hope to contribute effectively.

**Reply by arnoldsu** [reply]
I will be there. but I am too late to register; Romain or anyone who can email me the video link?

**Comment by alistairsellar** [comment]
For anyone looking for light reading for their journey to the workshop (or bedtime reading if you are joining remotely), here is the draft strategy document: https://github.com/ESMValGroup/Community/discussions/164#discussioncomment-9546505. We'll discuss this on Wednesday morning, and comments welcome in the meantime (in the doc please).

**Comment by glpotter** [comment]
Sorry, but I won’t  be able to  make much of the virtual meeting. Please send out the invitation link again. 

Thanks,
Jerry

**Comment by alistairsellar** [comment]
Here is the folder that @bouweandela just mentioned: https://github.com/ESMValGroup/Community/tree/main/Workshops/2024-05.

If you give a presentation at this workshop, please open a PR to add your presentation to this folder (ask for help if you're not sure how). If your presentation contains something you don't want to be public, please edit that out and share the rest.

**Comment by rswamina** [comment]
Please add to and edit notes on the observation data discussion from today in the [discussion document](https://docs.google.com/document/d/1P3QcODeu82IsakYwBUXbzytk8xdRV2_-PlMRBzmN4w0/edit).
